content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,225 @@
+    <Caption>Separate auto-saving controls from an explicitly saved form</Caption>
+  </Do>
+  <Dont>
+    <img alt="A form with a single section and it contains a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887936-d41900ca-2d63-43ac-bf82-55b4bba0df61.png" />
+    <Caption>Mix auto-saving controls with explictly saved controls in the same form</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Unsaved changes
+
+With the addition of the command palette there are additional opportunities to accidentally navigate away from a page with unpublished edits.
+
+If a user tries to navigate away from a page with a form that has unsaved changes, you can optionally warn them that their changes will be lost using the [`beforeunload` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). Modern browsers prevent developers from customizing the `alert` message, so we're limited to a generic message like "Do you want to leave this site? Changes you made may not be saved."
+
+### Declarative controls
+
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Browser-native `<select>` dropdown menus
+- Multi-select dropdown menus
+
+To maintain consistent interaction patterns across your app, do not use segmented controls in an explicitly saved form. These should only be used with automatic saving.
+
+To ensure your forms are accessible, do not automatically save declarative inputs.
+
+#### Problems with autosaving declarative inputs:
+**Text inputs**
+- Users are expecting to set a value and then submit
+- Sensitive information could be unintentionally submitted. For example: a change password input
+
+**Checkbox groups**
+- It could be unclear whether or not their selections have been saved
+- Users may accidentally submit a selection that they clicked by accident
+
+**Radio button groups**
+- Screenreader users can't read the options without selecting them. This could accidentally apply a selection the user didn't want.
+
+**Browser-native `<select>` dropdowns**
+- A select dropdown has a similar problem as a radio button group. Options are selected when a user focuses the select and uses their arrow keys to read through each option
+
+<DoDontContainer>
+  <Do>
+    <img src="https://user-images.githubusercontent.com/2313998/151623711-ad294946-fd67-428d-96b2-e4f534857c8a.png" />
+    <Caption>Use save and cancel buttons in dropdowns that take multiple selections</Caption>
+  </Do>
+  <Dont>
+    <img src="https://user-images.githubusercontent.com/2313998/151623713-12028e6d-86ba-4ae6-b46b-d3c88c0fa2c0.png" />
+    <Caption>Automatically save selections when the dropdown is closed</Caption>
+  </Dont>
+</DoDontContainer>
+
+## Automatic saving
+
+Automatic saving applies changes as they are made.
+
+Automatic saving should be used when the user expects instant feedback from the change they made. Changes should only be saved automatically using [imperative inputs](#imperative-inputs).
```

Comment:
Good catch! It will not be added back in this PR, so I'll remove it. The imperative inputs section will be after we have more imperative input components (for example, toggle switch and segmented control).